spec/app/domain/authentication/authenticate_spec.rb

```diff
@@ -0,0 +1,172 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+RSpec.describe 'Authentication::Authenticate' do
+  ####################################
+  # Available Authenticators - doubles
+  ####################################
+
+  def authenticator(pass:)
+    double('Authenticator').tap do |x|
+      allow(x).to receive(:valid?).and_return(pass)
+    end
+  end
+
+  def input(
+    authenticator_name: 'authn-always-pass',
+    service_id: nil,
+    account: 'my-acct',
+    username: 'my-user',
+    password: 'my-pw',
+    origin: '127.0.0.1',
+    request: nil
+  )
+    Authentication::Input.new(
+      authenticator_name: authenticator_name,
+      service_id: service_id,
+      account: account,
+      username: username,
+      password: password,
+      origin: origin,
+      request: request
+    )
+  end
+
```

Comment:
I don't think this abstraction buys us anything.   Wherever you would currently write:
```
authenticators['authn-always-pass']
```
just write:
```
authenticator(pass: true)
```
It's equally readable and clear.

Reply:
Fun fact: it's your code :)

Anyway, this is needed as the line you mention (`authenticators['authn-always-pass']`) is not on the test code but in prod. We just pass `authenticators` as input to the `Authentication::Authenticate` CommandClass.

Let me know if I'm wrong.

Reply:
> Fun fact: it's your code :)

Reply:
Ignore the above comment.  Per our slack convo, the `authenticators['authn-always-pass']` stuff is fine as is.